README.md

```diff
@@ -45,8 +45,36 @@ returns the length of the shortest tour (not the tour itself).
 
 Test your new function; I've provided some basic testing code in `code.test.js`.
 
+#### Stoping Criterion:
+The stopping criterion I choose is n * n where n is the number of cities in the given distance_matrix. This ensures that as the matrix gets bigger the stopping criterion is increased to find the best route. while going through the iterations the code checks if there is any improvment in the route found so far. If there is improvement it updates the best route and sets the current route to it. if no improvment found it only sets the new route to be current route and continues through the iterations
+
+#### Choosing i and k:
+I choose to select i randomly between 0 and (distance_matrix.length - 1) and k is choosen randomly ensuring that it is greater than i so we dont end up making the same swapped list again.
+
 ## Runtime Analysis
 
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+
+The worst case time complexity for the code I did is $O(n^3)$. This is because the outer loop runs a maximum of $n * n$ iterations and the calculateRoute is called in each iteration which takes about n time complexity. Together the total time complexity is $O(n^3)$
+
+The worst case memory complexity for the code implemented is $O(n)$. This is because bestroute, currentroute, newroute all are arrays of size n, which are stored until we find the best route. In total they add up to 3n. Asymptotically we ignore constants. So the total worst case memory complexity is $O(n)$ 
+
+
+References Used:
+
+Pseudo code provided in the question was very helpful in getting started. I was able to figure most of the helper functions like genraterandom, calculateRoute, twooptswap. The main part of the code was also easy, however I struggled in handling the edge cases where the tests rans forever.
+
+I looked at the following repo to get help with the edge cases:
+
+tsp-local-search-swilso59
+
+The following repo helped me handle the stopping criterion. I originally used having like a max iterations until no improvment found. But this made the test run forever. 
+
+tsp-local-search-ClaytonBrown4741
+
+I have also used ChatGPT to help me figure out where I messed up for some of the tries I did where i made a silly mistake calling the calculateRoute funtion without the distance_matrix being passed to it
```

Comment:
So what part of the code was written by you and what came from other sources?

Reply:
I have updated the readme file.